### PR TITLE
Add Local BTV m3u file

### PR DESCRIPTION
**Pull request by MapGuy11**
1. Add Philadelphia LocalBTV Channels
2. Create a new file just for LocalBTV

### CODE REVIEW

**Comment by MapGuy11** [comment]
@LaneSh4d0w normal web browser works but VLC seems to not work can you look at it?

**Review by Sphinxroot — APPROVED** [review]
all work on kodi with the proper  format

**Comment by Central302** [comment]
Wired, Working on GSE Smart IPTV Player via iPhone.

**Comment by LaneSh4d0w** [comment]
> @LaneSh4d0w normal web browser works but VLC seems to not work can you look at it?

VLC thinks it's your server streaming the data, it seems there's no full redirection to btv whatsoever.

**Comment by LaneSh4d0w** [comment]
However, I think I see what you tried. Once you generate the M3U file, be sure that the links inside are redirecting to their server.

File: channels/us_localbtv.m3u [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#EXTM3U
+#EXTINF:-1 tvg-id="WDPNDT2.us" tvg-country="US" tvg-language="English" tvg-logo="https://upload.wikimedia.org/wikipedia/en/b/be/Grit_TV_Network_Logo.gif" user-agent="Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36" group-title="Entertainment",WDPN-DT2 (Grit)
+#EXTVLCOPT:http-user-agent=Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
+https://v-pi.theus6tv.tk/hls/2.2/playlist.m3u8
+#EXTINF:-1 tvg-id="WDPNDT5.us" tvg-country="US" tvg-language="English" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/f/f6/RetroTV_logo.png" user-agent="Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36" group-title="Entertainment",WDPN-DT5 (Retro TV)
+#EXTVLCOPT:http-user-agent=Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
+https://v-pi.theus6tv.tk/hls/2.5/playlist.m3u8
```
[the remark this answers is not in the document]
Comment (MapGuy11) [reply channels/us_localbtv.m3u:7]: Yeah, they stream ShopLC at night.

**Comment by MapGuy11** [comment]
> However, I think I see what you tried. Once you generate the M3U file, be sure that the links inside are redirecting to their server.
Here is one of the playlist.m3u8 files:

#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:BANDWIDTH=2000,RESOLUTION=768x432
https://v-pi.theus6tv.tk/main.php?channel=WFMZSD
#EXT-X-STREAM-INF:BANDWIDTH=4500,RESOLUTION=1280x720
https://v-pi.theus6tv.tk/main.php?channel=WFMZHD

**Comment by MapGuy11** [comment]
@LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.

Copy of VLC error code:

adaptive error: Failed to create demuxer 0000000000000000 Unknown
adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found

**Comment by LaneSh4d0w** [comment]
> @LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.
> 
> Copy of VLC error code:
> 
> adaptive error: Failed to create demuxer 0000000000000000 Unknown adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found

Maybe try to play with the location header.

**Comment by MapGuy11** [comment]
> > @LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.
> > Copy of VLC error code:
> > adaptive error: Failed to create demuxer 0000000000000000 Unknown adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found
> 
> Maybe try to play with the location header.

what is the location header?

**Comment by LaneSh4d0w** [comment]
> > > @LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.
> > > Copy of VLC error code:
> > > adaptive error: Failed to create demuxer 0000000000000000 Unknown adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found
> > 
> > 
> > Maybe try to play with the location header.
> 
> what is the location header?

When you make a 301 / 302 redirection code, the browser reads the "Location" header from the response headers, to see where the browser should redirect to. It's basically telling directions.

**Comment by MapGuy11** [comment]
> > > > @LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.
> > > > Copy of VLC error code:
> > > > adaptive error: Failed to create demuxer 0000000000000000 Unknown adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found
> > > 
> > > 
> > > Maybe try to play with the location header.
> > 
> > 
> > what is the location header?
> 
> When you make a 301 / 302 redirection code, the browser reads the "Location" header from the response headers, to see where the browser should redirect to. It's basically telling directions.

Ok I'll see what I can do with PHP and nginx.

**Comment by MapGuy11** [comment]
> > > > > @LaneSh4d0w I figured it out I don't know why but vlc doesn't get the .m3u8 file after the PHP script redirects it to btv. It thinks the .ts files are on the local server let me see if I can do something about that.
> > > > > Copy of VLC error code:
> > > > > adaptive error: Failed to create demuxer 0000000000000000 Unknown adaptive error: Failed reading https://v-pi.theus6tv.tk:443/768x432/1642418045.ts: HTTP/1.1 404 Not Found
> > > > 
> > > > 
> > > > Maybe try to play with the location header.
> > > 
> > > 
> > > what is the location header?
> > 
> > 
> > When you make a 301 / 302 redirection code, the browser reads the "Location" header from the response headers, to see where the browser should redirect to. It's basically telling directions.
> 
> Ok I'll see what I can do with PHP and nginx.

@LaneSh4d0w it's already sending a proper location redirect header. Wierd

**Comment by MapGuy11** [comment]
@LaneSh4d0w Yeah everything is correct I can't seem to get vlc to understand.

**Review by BellezaEmporium — APPROVED** [review]

**Review by freearhey — APPROVED** [review]

**Comment by MapGuy11** [comment]
@freearhey Thanks Everybody!

**Comment by vlorty** [comment]
If you can be so kind to get the rest? We can use the Phoenix, San Francisco etc. is there a link for the rest?

**Comment by LaneSh4d0w** [comment]
@MapGuy11 this is what I thought, VLC dislikes double redirections.

Use the link you've embedded in the playlist.m3u8 file you've put into the playlist files, and it'll work.

**Pull request merged**